### PR TITLE
Add docs for failed_data_files_count

### PR DESCRIPTION
**Pull request by dacort**
(unfiled) Noticed while running a `rewrite_data_files` job that there was an extra field in my output that's not present in the docs.

### CODE REVIEW

File: docs/spark-procedures.md [hunk 1]
```diff
@@ -360,6 +360,7 @@ Iceberg can compact data files in parallel using Spark with the `rewriteDataFile
 | `rewritten_data_files_count` | int | Number of data which were re-written by this command |
 | `added_data_files_count`     | int | Number of new data files which were written by this command |
 | `rewritten_bytes_count`      | long | Number of bytes which were written by this command |
+| `failed_data_files_count`    | int | Number of data files that failed to write if `partial-progress.enabled` is true and any rewrites failed |
```
Comment (bitsondatadev) [inline docs/spark-procedures.md:363]:
```suggestion
| `failed_data_files_count`    | int | Number of data files that failed to write if `partial-progress.enabled` is true |
```

I don't think the last bit is necessary if the count is 0 if none failed.

**Review by bitsondatadev — APPROVED** [review]
one nit otherwise LGTM

**Review by bitsondatadev — APPROVED** [review]
buenisimo! can you squash the commits please?

**Comment by dacort** [comment]
Squashed!

File: docs/spark-procedures.md [hunk 2]
```diff
@@ -360,6 +360,7 @@ Iceberg can compact data files in parallel using Spark with the `rewriteDataFile
 | `rewritten_data_files_count` | int | Number of data which were re-written by this command |
 | `added_data_files_count`     | int | Number of new data files which were written by this command |
 | `rewritten_bytes_count`      | long | Number of bytes which were written by this command |
+| `failed_data_files_count`    | int | Number of data files that failed to write if `partial-progress.enabled` is true |
```
Comment (nastra) [inline docs/spark-procedures.md:363]:
```suggestion
| `failed_data_files_count`    | int | Number of data files that failed to be rewritten when `partial-progress.enabled` is true |
```

**Review by rdblue — APPROVED** [review]

**Comment by rdblue** [comment]
Thanks, @dacort!

**Pull request merged**